scripts/generateForDay.sh

```diff
@@ -1,41 +1,168 @@
+if [ -z "${ALLSKY_HOME}" ] ; then
+	export ALLSKY_HOME=$(realpath $(dirname "${BASH_ARGV0}")/..)
+fi
+
+# If the first argument is "--upload" then we upload files, otherwise we create them.
+if [ "$1" = "--upload" ]; then
+	shift
+	TYPE="UPLOAD"
+	MSG1="upload"
+	MSG2="uploaded"
+else
+	TYPE="GENERATE"
+	MSG1="create"
+	MSG2="created"
+fi
+
+source "${ALLSKY_HOME}/variables.sh"
+source "${ALLSKY_CONFIG}/config.sh"
+source "${ALLSKY_SCRIPTS}/filename.sh"
+[ "${TYPE}" = "UPLOAD" ] && source "${ALLSKY_SCRIPTS}/ftp-settings.sh"
 
-RED='\033[0;31m'
-GREEN='\033[0;32m'
-NC='\033[0m' # No Color
 
-if [ $# -lt 1 ]
-  then
-    echo -en "${ME}: ${RED}You need to pass a day argument\n"
-        echo -en "    ex: ${ME} 20180119${NC}\n"
-        exit 3
+if [ $# -eq 0 -o "${1}" = "-h" -o "${1}" = "--help" ]; then
+	echo -en "${RED}"
+	echo -e "Usage: ${ME} [-k] [-s] [-t] DATE"
+	echo -e "    where:"
+	echo -e "      'DATE' is the day in '${ALLSKY_IMAGES}' to process"
+	echo -e "      'k' is to ${MSG1} a keogram"
+	echo -e "      's' is to ${MSG1} a startrail"
+	echo -e "      't' is to ${MSG1} a timelapse"
+	echo -e "    If you don't specify k, s, or t, all three will be ${MSG2}."
+	echo -e "${NC}"
+	exit 0
 fi
 
-DATE=$1
-DIR="$ALLSKY_HOME/images/$DATE"
+if [ $# -gt 1 ] ; then
+	DO_K="false"
+	DO_S="false"
+	DO_T="false"
+	while [ $# -gt 1 ]
+	do
+		if [ "${1}" = "-k" ] ; then
+			DO_K="true"
+		elif [ "${1}" = "-s" ] ; then
+			DO_S="true"
+		elif [ "${1}" = "-t" ] ; then
+			DO_T="true"
+		else
+			echo "Unknown image type: '${1}'; ignoring."
+		fi
+		shift
+	done
+else
+	DO_K="true"
+	DO_S="true"
+	DO_T="true"
+fi
+
+DATE="${1}"
+DATE_DIR="${ALLSKY_IMAGES}/${DATE}"
+if [ ! -d "${DATE_DIR}" ] ; then
+	echo -e "${ME}: ${RED}ERROR: '${DATE_DIR}' not found!${NC}"
+	exit 2
+fi
 
-# Generate timelapse from collected images
-echo -e "${ME}: Generating Keogram\n"
-mkdir -p ${DIR}/keogram/
-../keogram ${DIR}/ $EXTENSION ${DIR}/keogram/keogram-$DATE.jpg
-echo -e "\n"
+if [ "${DO_K}" = "true" ] ; then
+	KEOGRAM_FILE="keogram-${DATE}.${EXTENSION}"
+	UPLOAD_FILE="${DATE_DIR}/keogram/${KEOGRAM_FILE}"
+	if [ "${TYPE}" = "GENERATE" ]; then
+		# Don't include ${ME} since this script is always executed manually
+		echo -e "===== Generating Keogram"
+		mkdir -p "${DATE_DIR}/keogram"
 
-# Generate startrails from collected images. Treshold set to 0.1 by default in config.sh to avoid stacking over-exposed images
-echo -e "${ME}: Generating Startrails\n"
-mkdir -p ${DIR}/startrails
-../startrails ${DIR}/ $EXTENSION $BRIGHTNESS_THRESHOLD ${DIR}/startrails/startrails-$DATE.jpg
-echo -e "\n"
+		# The keogram command outputs one line for each of the many hundreds of files,
+		# and this adds needless clutter to the output, so send output to a tmp file so we
+		# can output the number of images.
+		TMP="${ALLSKY_TMP}/keogramTMP.txt"
+		"${ALLSKY_HOME}/keogram" -d "${DATE_DIR}" -e ${EXTENSION} -o "${UPLOAD_FILE}" ${KEOGRAM_PARAMETERS} > "${TMP}"
```

Comment:
Keogram now runs quietly, so you don't need to send the output to a temp file. But you can give it the `-v` flag to enable the trace output if you think there is value in capturing it.

Reply:
No longer uses TMP file.  Not worth using `-v`.